sympy/matrices/matrices.py

```diff
@@ -189,6 +189,9 @@ def __neg__(self):
         return -1*self
 
     def _matrix_pow_by_jordan_blocks(self, num):
+        if self.det() == 0 and num < 0:
```

Comment:
Just one minor detail: `num < 0` before `self.det() == 0`, as it is faster to evaluated. This means the determinant will only be calculated if `num < 0`.